src/EntityComponentManager.cc

```diff
@@ -1257,6 +1282,7 @@ void EntityComponentManager::SetState(
         // values.
         // igndbg << *comp << "  " << *newComp.get() << std::endl;
         // *comp = *newComp.get();
+        this->dataPtr->AddModifiedComponent(entity);
```

Comment:
is this call needed here? Given that the above msg says internal error, I expect that something is wrong is it gets to this else block. Looking at the code, the component value is just being updated by removing the old component and creating a new one?

Reply:
It's not needed currently but I put the call there in case the above `TODO` is addressed. Should I comment it out with a note to add it in when the `TODO` is addressed and if `SetChanged` is not called?

Reply:
I see, yeah you can comment it out and add a note here

Reply:
Done 70c432050ff7c18d5cb3f79de5cb697ce950ff14